### DISCUSSION TITLE
Shader not working

### DISCUSSION THREAD

**Discussion by THEkatinamicrowave** [CodenameCrew/CodenameEngine#451] [Q&A] [answered]
here's what the shader is supposed to do

![image](https://github.com/user-attachments/assets/fb6cdb0f-c0e7-4a6c-8134-ec3e1710e68d)
The goal of the shader is to save on sprite-making necessity and improve character-port-ability. Here's the ideal rundown:

1. Grab a pixel
2. If the pixel's alpha is less than 1.0, do this
3. Make a 1:4 ratio rectangle (for now it's hardcoded to be 2x8) with the top left corner centered on the pixel
4. Make everything within the rectangle red, the intensity corresponding to the pixel's alpha
5. If the pixel's alpha is 1.0, then no light would get through anyway, so there's no need to change shit

here's the code

```glsl
#pragma header

uniform sampler2D texture1;
varying vec2 TexCoords;

void main()
{
    vec4 dumbasscolor = texture2D(texture1, TexCoords);
    
    if (dumbasscolor.a < 1.0)
    {
        vec2 pixelSize = vec2(1.0 / textureSize2D(texture1, 0).x, 1.0 / textureSize2D(texture1, 0).y);
        for (int x = 0; x < 2; ++x)
        {
            for (int y = 0; y < 8; ++y)
            {
                vec2 coordinates = TexCoords + vec2(x * pixelSize.x, y * pixelSize.y);
                vec4 newshit = texture2D(texture1, coordinates);
                newshit.rgb = mix(newshit.rgb, vec3(1.0, 0.0, 0.0), dumbasscolor.a);
                gl_FragColor = newshit;
            }
        }
    }
    else
    {
        gl_FragColor = dumbasscolor;
    }
}
```
I'm not even gonna bother demonstrating the implementation because I know it's right
anyway, the shader isn't doing rat shit

**Comment by SrtHero278** [comment, marked as answer]
hate to break it to you but
1. textureSize doesnt exist in flixel (textureSize(texture1) -> openfl_TextureSize)
2. you are using the wrong variable names. (texture1 -> bitmap, TexCoords -> openfl_TextureCoordv)

**Reply by THEkatinamicrowave** [reply]
this is correct but apparently this whole shader is just freaking like idk it just doesn't work and is really tedious
